napari/utils/_register.py

```diff
@@ -56,7 +56,8 @@ def create_func(cls, name=None, doc=None):
     )
 
     execdict = {cls_name: cls, 'napari': sys.modules.get('napari')}
-    exec(src, execdict)
+    code = compile(src, filename=filename, mode='exec')
```

Comment:
this change just improves the stack trace, showing the proper filename instead of `"<string>"`